v2.4-dev: Crash when closing the window

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Running a Fyne application with the `systray` feature running makes the application crash right after the window close button has been pressed. I noticed that in `fyne_demo`. Rymdport, which doesn't use the system tray (yet), does not experience this issue.

```
2023/07/19 18:40:25 Lifecycle: Exited Foreground
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xa749d5]

goroutine 82 [running]:
fyne.io/systray.stayRegistered()
	/home/jacob/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230717104455-e8b2022f921f/systray_unix.go:269 +0x255
created by fyne.io/systray.nativeStart
	/home/jacob/go/pkg/mod/fyne.io/systray@v1.10.1-0.20230717104455-e8b2022f921f/systray_unix.go:230 +0x885

```

### How to reproduce

1. Open `fyne_demo`
2. Close the window using the close button

### Screenshots

_No response_

### Example code

Use `fyne_demo` for reproducing.

### Fyne version

940dcdc82b574a33cff85e71b6191582b2a1fb54

### Go compiler version

1.20.6

### Operating system and version

Fedora Linux 38

### Additional Information

I didn't do any triaging but I suspect that it may be caused by changes introduced in dd3155cab9e63839560fa383daaade6a5e1312c4.

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module fyne.io/fyne/v2
 go 1.17
 
 require (
-	fyne.io/systray v1.10.1-0.20230717104455-e8b2022f921f
+	fyne.io/systray v1.10.1-0.20230722100817-88df1e0ffa9a
 	github.com/BurntSushi/toml v1.1.0
 	github.com/fogleman/gg v1.3.0
 	github.com/fredbi/uri v0.1.0

